C7GameData/AIData/TilePath.cs

```diff
@@ -37,5 +74,22 @@ public static TilePath EmptyPath(Tile destination) {
 			return new TilePath(destination, new Queue<Tile>());
 		}
 
+		public static float getMovementCost(Tile from, TileDirection dir, Tile newLocation) {
+			// River crossings disrupt roads, so check that first.
+			if (from.HasRiverCrossing(dir)) return newLocation.MovementCost();
+
+			// Travelling between two tiles with railroads is free.
```

Comment:
shouldn't this logic come first?

Reply:
I had to test this and technically no, if you don't have the "allows bridges" tech, crossing a river on a railroad does use movement! The real omission is checking for that flag, but nether it nor railroads are implemented yet.
Good to see you back @Sean-Brown :+1:

Reply:
interesting, I assumed that if you have railroads then you have a bridge crossing tech